src/cryptsetup/cryptsetup-generator.c

```diff
@@ -50,10 +50,12 @@ static int split_keyspec(const char *keyspec, char **ret_keyfile, char **ret_key
         _cleanup_free_ char *keyfile = NULL, *keydev = NULL;
         const char *c;
 
-        assert(keyspec);
         assert(ret_keyfile);
         assert(ret_keydev);
 
+        if (!keyspec)
+                return 0;
```

Comment:
following our usual logic that on success we should initialize all return parameters I think this should be:

```c
if (!keyspec) {
        *ret_keyfile = *ret_keydev = NULL;
        return 0;
}
```

Reply:
Righto. Updated.